gtsam/hybrid/HybridConditional.cpp

```diff
@@ -100,79 +99,68 @@ bool HybridConditional::equals(const HybridFactor &other, double tol) const {
   if (auto gm = asHybrid()) {
     auto other = e->asHybrid();
     return other != nullptr && gm->equals(*other, tol);
-  }
-  if (auto gc = asGaussian()) {
+  } else if (auto gc = asGaussian()) {
     auto other = e->asGaussian();
     return other != nullptr && gc->equals(*other, tol);
-  }
-  if (auto dc = asDiscrete()) {
+  } else if (auto dc = asDiscrete()) {
     auto other = e->asDiscrete();
     return other != nullptr && dc->equals(*other, tol);
-  }
-
-  return inner_ ? (e->inner_ ? inner_->equals(*(e->inner_), tol) : false)
-                : !(e->inner_);
+  } else
+    return inner_ ? (e->inner_ ? inner_->equals(*(e->inner_), tol) : false)
+                  : !(e->inner_);
 }
 
 /* ************************************************************************ */
 double HybridConditional::error(const HybridValues &values) const {
   if (auto gc = asGaussian()) {
     return gc->error(values.continuous());
-  }
-  if (auto gm = asHybrid()) {
+  } else if (auto gm = asHybrid()) {
     return gm->error(values);
-  }
-  if (auto dc = asDiscrete()) {
+  } else if (auto dc = asDiscrete()) {
     return dc->error(values.discrete());
-  }
-  throw std::runtime_error(
-      "HybridConditional::error: conditional type not handled");
+  } else
+    throw std::runtime_error(
+        "HybridConditional::error: conditional type not handled");
 }
 
 /* ************************************************************************ */
 AlgebraicDecisionTree<Key> HybridConditional::errorTree(
     const VectorValues &values) const {
   if (auto gc = asGaussian()) {
-    return AlgebraicDecisionTree<Key>(gc->error(values));
-  }
-  if (auto gm = asHybrid()) {
+    return {gc->error(values)};  // NOTE: a "constant" tree
+  } else if (auto gm = asHybrid()) {
     return gm->errorTree(values);
-  }
-  if (auto dc = asDiscrete()) {
-    return AlgebraicDecisionTree<Key>(0.0);
-  }
-  throw std::runtime_error(
-      "HybridConditional::error: conditional type not handled");
+  } else if (auto dc = asDiscrete()) {
```

Comment:
Bug 1

Reply:
Oh interesting, I was missing the `DiscreteFactor::errorTree` method. There's actually an improvement here I can make.